src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarCommercialButtons.tsx

```diff
@@ -354,15 +355,64 @@ export const ArtworkSidebarCommercialButtons: React.FC<ArtworkSidebarCommercialB
     artwork.isAcquireable || activePartnerOffer
   )
 
+  const SaleMessageOrOfferDisplay: FC = () => {
+    if (activePartnerOffer) {
+      return (
+        <>
+          <Spacer y={2} />
+          <Flex>
+            <Text
+              variant="xs"
+              color="blue100"
+              backgroundColor="blue10"
+              px={0.5}
+            >
+              Limited-Time Offer
+            </Text>
+          </Flex>
+
+          <Spacer y={0.5} />
+
+          <Flex
+            flexDirection={["column", "row"]}
+            width="100%"
+            justifyContent="start"
+          >
+            <SaleMessage
+              saleMessage={activePartnerOffer.priceWithDiscount?.display}
```

Comment:
Non-blocking: Should never happen, but it strikes me here that this could technically be not present.

Reply:
You mean `activePartnerOffer` ? This part is only rendered if it's present according to the check above. I guess I need to be checking `partnerOffer` as you noted above. And price should always be present there. This `?` is just for error cases so that page does not crash.